driver/src/price_finding/stablex_naive_solver.rs

```diff
@@ -0,0 +1,242 @@
+use crate::price_finding::error::PriceFindingError;
+use crate::price_finding::PriceFinding;
+use crate::util;
+use dfusion_core::models::{AccountState, Order, Solution};
+use std::cmp::Ordering;
+
+/// A naive solver for the stableX contract. It works almost identically to the
+/// Snapp naive solver except it has a few extra requirements and it doesn't try
+/// to be as optimal.
+///
+/// The solver works by finding the first pair of orders which matching sell and
+/// buy tokens, where one of the traded tokens is the fee token, whose users have
+/// enough balance to fully fill, and that has a viable solution.
+///
+/// The solution is computed by taking the price of the larger order and and
+/// completely filling the smaller order. A large order is defined by having
+/// more non-fee token volume. Account balances are checked for each solution.
+pub struct StableXNaiveSolver;
+
+impl PriceFinding for StableXNaiveSolver {
```

Comment:
Have you tried if plugging this implementation into the Snapp e2e tests will work? I don't see why it shouldn't

Reply:
I did not. But after discussing with @bh2smith and @josojo I got a little scared. The plan is to merge these two solvers and just use this implementation to be able to get e2e tests passing so other PRs could be merged.

I want to work on merging the naive solvers this week but need to understand snapp a bit more before I can.